### PR TITLE
Allegations of harm, C7 submission and respondent doc management

### PR DESCRIPTION
**Pull request by anshikanigam1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-5112

### Change description ###
Allegations of harm, C7 submission and respondent doc management

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/steps/common/safety-concerns/child/report-abuse/content.ts [hunk 1]
```diff
@@ -248,16 +256,35 @@ export const form: FormContent = {
...
 };
 
-export const getFormFields = (caseData: Partial<CaseWithId>, abuseType: C1AAbuseTypes): FormContent => {
+export const getFormFields = (
+  caseData: Partial<CaseWithId>,
+  abuseType: C1AAbuseTypes,
+  C100RebuildJourney
+): FormContent => {
   const sessionData: C1ASafteyConcernsAbuse = caseData?.c1A_safteyConcerns?.child?.[abuseType];
+  const sessionChildrenData = caseData?.cd_children ?? [];
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/child/report-abuse/content.ts:267]: c100Children

File: src/main/steps/common/safety-concerns/child/report-abuse/content.ts [hunk 2]
```diff
@@ -248,16 +256,35 @@ export const form: FormContent = {
...
+  const sessionChildrenData1 = caseData?.newChildDetails ?? [];
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/child/report-abuse/content.ts:268]: CARespChildren

File: src/main/steps/common/safety-concerns/child/report-abuse/content.ts [hunk 3]
```diff
@@ -248,16 +256,35 @@ export const form: FormContent = {
...
+
+  const data1: Childinfo[] = sessionChildrenData.map(i => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/child/report-abuse/content.ts:270]: c100ChildrenDetails

File: src/main/steps/common/safety-concerns/child/report-abuse/content.ts [hunk 4]
```diff
@@ -248,16 +256,35 @@ export const form: FormContent = {
...
+    return {
+      id: i.id,
+      firstName: i.firstName,
+      lastName: i.lastName,
+    };
+  });
+
+  const data2: Childinfo[] = sessionChildrenData1.map(i => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/child/report-abuse/content.ts:278]: CARespChildrenDetails

File: src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    c1A_passportOffice,
+    c1A_abductionPassportOfficeNotified,
+    c1A_previousAbductionsShortDesc,
+    c1A_policeOrInvestigatorInvolved,
+    c1A_policeOrInvestigatorOtherDetails,
+    c1A_childAbductedBefore,
+    c1A_safteyConcerns,
+  } = caseData;
+
+  const data = {
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts:32]: validate. looks duplicate

File: src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts [hunk 6]
```diff
@@ -0,0 +1,224 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Case } from '../../../../app/case/case';
+import { C1AAbuseTypes, C1ASafteyConcernsAbout, YesOrNo } from '../../../../app/case/definition';
+
+export const transformCaseData = (caseData: Partial<Case>): string => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts:5]: prepareRequest

File: src/main/steps/tasklistresponse/responseCaseSequence.ts [hunk 7]
```diff
@@ -263,135 +257,135 @@ export const responseCaseSequence: Step[] = [
...
-    getNextStep: () => C1A_SAFETY_CONCERNS_CHECK_YOUR_ANSWERS_SAVE,
-  },
-  {
-    url: PRL_C1A_SAFETY_CONCERNS_NOFEEDBACK,
-    showInSection: Sections.AboutRespondentCase,
-    getNextStep: () => PRL_C1A_SAFETY_CONCERNS_CONCERNS_ABOUT_CHILD,
-  },
+    getNextStep: () => applyParms(C1A_SAFETY_CONCERNS_CONCERN_GUIDANCE, { root: RootContext.RESPONDENT }) as PageLink,
+  },
+  // {
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/responseCaseSequence.ts:262]: validate and remove

File: src/main/steps/tasklistresponse/utils.ts [hunk 8]
```diff
@@ -40,24 +39,27 @@ export const mapDataInSession = (userCase: CaseWithId, userId: UserDetails['id']
...
-
-const setDataInSession = (userCase: CaseWithId, partyDetails: PartyDetails): void => {
-  if (partyDetails?.response?.safetyConcerns) {
-    Object.assign(userCase, mapSafetyConcernsDetails(partyDetails));
+function setDataInSession(userCase: CaseWithId, partyDetails: PartyDetails) {
+  if (partyDetails?.response) {
+    //Object.assign(userCase, { ...mapSafetyConcernsDetails(partyDetails), ...userCase });
+    if (partyDetails?.response.respondingCitizenAoH) {
+      const obj = JSON.parse(partyDetails?.response.respondingCitizenAoH);
+      Object.assign(userCase, obj);
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/utils.ts:47]: parse it in same line

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/tasklistresponse/utils.ts [hunk 9]
```diff
@@ -40,24 +39,26 @@ export const mapDataInSession = (userCase: CaseWithId, userId: UserDetails['id']
...
   }
 };
-
-const setDataInSession = (userCase: CaseWithId, partyDetails: PartyDetails): void => {
-  if (partyDetails?.response?.safetyConcerns) {
-    Object.assign(userCase, mapSafetyConcernsDetails(partyDetails));
+function setDataInSession(userCase: CaseWithId, partyDetails: PartyDetails) {
+  if (partyDetails?.response) {
+    //Object.assign(userCase, { ...mapSafetyConcernsDetails(partyDetails), ...userCase });
+    if (partyDetails?.response.respondingCitizenAoH) {
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/utils.ts:45]: u can use _ get method to check and keep the conditional check simple

File: src/main/steps/urls.ts [hunk 10]
```diff
@@ -382,31 +382,37 @@ export const PAYMENT_GATEWAY_ENTRY_URL: PageLink = '/payments/gateway';
...
-export const C100_C1A_SAFETY_CONCERNS_REPORT_CHILD_ABUSE: PageLink = `${C100_C1A_SAFETY_CONCERNS_CHILD}/report-abuse/:abuseType`;
-export const C100_C1A_SAFETY_CONCERNS_CONCERNS_ABOUT_APPLICANT: PageLink = `${C100_C1A_SAFETY_CONCERNS_APPLICANT}/concerns-about`;
-export const C100_C1A_SAFETY_CONCERNS_REPORT_APPLICANT_ABUSE: PageLink = `${C100_C1A_SAFETY_CONCERNS_APPLICANT}/report-abuse/:abuseType`;
-export const C100_C1A_SAFETY_CONCERNS_OTHER_CONCERNS_DRUGS: PageLink = `${C100_C1A_SAFETY_CONCERNS}/other-concerns/drugs`;
-export const C100_C1A_SAFETY_CONCERNS_ORDERS_REQUIRED_UNSUPERVISED: PageLink = `${C100_C1A_SAFETY_CONCERNS}/orders-required/unsupervised`;
-export const C100_C1A_SAFETY_CONCERNS_OTHER: PageLink = `${C100_C1A_SAFETY_CONCERNS}/other-concerns/other-issues`;
-export const C100_C1A_SAFETY_CONCERNS_ORDERS_REQUIRED_COURT_ACTION: PageLink = `${C100_C1A_SAFETY_CONCERNS}/orders-required/court-action`;
-export const C100_C1A_SAFETY_CONCERNS_NOFEEDBACK: PageLink = `${C100_C1A_SAFETY_CONCERNS}/no-feedback`;
+/** common C1A Safety Concerns */
+export const C1A_SAFETY_CONCERNS: PageLink = '/:root/safety-concerns';
```
Comment (vivek-sekhar) [inline src/main/steps/urls.ts:386]: we need partyType to be added before root like const C1A_SAFETY_CONCERNS: PageLink = '/:partyType/:root/safety-concerns' and use this for below urls and not duplicate everytime.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/tasklistresponse/utils.ts (hunk 9 above)
Comment (sealDebolina) [reply src/main/steps/tasklistresponse/utils.ts]: addressed

File: src/main/steps/urls.ts (hunk 10 above)
Comment (sealDebolina) [reply src/main/steps/urls.ts:273]: @vivek-sekhar , As you have requested "/:root" for respondent url has been set to "respondondent" from"tasklistresponse" in definition.ts.

**Review by vivek-sekhar — APPROVED** [review]

**Review by AlokDatta — APPROVED** [review]

**Pull request merged**